sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -128,11 +130,39 @@ class QueryExecution(
     case _ => "command"
   }
 
+  private def assertNoStreamSourceMarkerNode(p: LogicalPlan): Unit = {
+    // In UnsupportedOperationChecker.checkForBatch, we check if the plan has any streaming node.
+    // That is more aggressive than just checking the marker node for streaming source which is
+    // yet to be materialized. We'd like to be a bit conservative here since this is the exact
+    // problematic case we figured out.
+    p.foreach {
```

Comment:
for leaf node command like `CreateDataSourceTableAsSelectCommand`, this check won't work?

Reply:
Yeah... LeafRunnableCommand doesn't seem to be something we could deal with it in general (as there is no general interface to look at underlying query). We could only deal with it when LeafRunnableCommand runs another query.